hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -302,19 +295,20 @@ public void close() {
   @Override
   public void reset() {
     try {
-      writeLock.lock();
+      viewLock.writeLock();
       clear();
       // Initialize with new Hoodie timeline.
       init(metaClient, getTimeline());
     } finally {
-      writeLock.unlock();
+      viewLock.writeUnlock();
     }
   }
 
   /**
    * Clear the resource.
    */
   protected void clear() {
+    assert viewLock.hasWriteLock();
```

Comment:
We can use `ValidationUtils.checkArgument` to check the result here.

e.g.,
ValidationUtils.checkArgument(viewLock.hasWriteLock(), "Current thread must hold the write lock on viewLock when clearing the resource");